app/components/signature-pill.js

```diff
@@ -11,84 +11,80 @@ export default class SignaturePillComponent extends Component {
-    } else if (this.isPrepared) {
-      return this.intl.t('sent');
-    } else if (this.isMarked) {
-      return this.intl.t('to-sign');
+  get skin() {
+    const { REFUSED, CANCELED } = constants.SIGNFLOW_STATUSES;
+    if (this.status.value === REFUSED || this.status.value === CANCELED) {
+      return 'error';
+    } else if (this.signingHubUrl.value) {
+      return 'link';
+    } else {
+      return 'ongoing';
     }
-    return "";
   }
 
-  loadData = task(async () => {
-    const signMarkingActivity = await this.args.signMarkingActivity;
-    if (signMarkingActivity) {
-      const signSubcase = await signMarkingActivity.signSubcase;
-      const signPreparationActivity = await signSubcase.signPreparationActivity;
-      const signApprovalActivities = await signSubcase.signApprovalActivities;
-      const signSigningActivities = await signSubcase.signSigningActivities;
-      const signCompletionActivity = await signSubcase.signCompletionActivity;
-      const signRefusalActivities = await signSubcase.signRefusalActivities;
-      const signCancellationActivity = await signSubcase.signCancellationActivity;
-
-      this.isMarked = !!signMarkingActivity;
-      this.isPrepared = !!signPreparationActivity;
-      this.hasToBeApproved = signApprovalActivities?.some((activity) => activity.startDate && !activity.endDate);
-      this.hasToBeSigned = signSigningActivities?.some((activity) => activity.startDate && !activity.endDate);
-      this.isSigned = !!signCompletionActivity;
-      this.isRefused = signRefusalActivities?.length;
-      this.isCancelled = !!signCancellationActivity;
+  get title() {
+    const {
+      MARKED,
+      PREPARED,
+      to_be_approved,
+      to_be_signed,
+      SIGNED,
+      REFUSED,
+      CANCELED,
+    } = constants.SIGNFLOW_STATUSES;
 
-      if (!this.isRefused) {
-        const piece = await this.args.piece;
-        const signFlow = await signSubcase.signFlow;
-        const signFlowCreator = await signFlow.creator;
-        const currentUser = this.currentSession.user;
-        if (piece && (signFlowCreator.id === currentUser.id) && !this.isSigned) {
-          const response = await fetch(
-            `/signing-flows/${signFlow.id}/pieces/${piece.id}/signinghub-url?collapse_panels=false`
-          );
-          if (response.ok) {
-            const result = await response.json();
-            this.signingHubUrl = result.url;
-          } 
-        }
-      }
+    switch (this.status.value) {
+      case MARKED:
+        return this.intl.t('to-sign');
+      case PREPARED:
+        return this.intl.t('sent');
+      case to_be_approved:
+        return this.intl.t('to-approve');
+      case to_be_signed:
+        return this.intl.t('to-sign');
+      case SIGNED:
+        return this.intl.t('signed');
+      case REFUSED:
+        return this.intl.t('refused');
+      case CANCELED:
+        return this.intl.t('cancelled');
+      default:
+        return '';
```

Comment:
We have a codelist with labels for this specific reason.

You'll need to get the `concept` resource for `this.status.uri` from the store for each status pill.
Those have a label.

Reply:
The translations can then be removed as well. (do check if they're used anywhere else)

Reply:
implemented in [3115039](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1787/commits/31150395fc58d04507ff7e8fadf6e97772e81e1d)